src/main/java/frc/team2412/robot/commands/intake/IntakeBitmapCommand.java

```diff
@@ -0,0 +1,98 @@
+package frc.team2412.robot.commands.intake;
+
+import static frc.team2412.robot.subsystem.IntakeSubsystem.IntakeConstants.*;
+import static frc.team2412.robot.subsystem.IndexSubsystem.IndexConstants.*;
+import static frc.team2412.robot.subsystem.ShooterSubsystem.ShooterConstants;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.team2412.robot.subsystem.IndexSubsystem;
+import frc.team2412.robot.subsystem.IntakeSubsystem;
+import frc.team2412.robot.subsystem.ShooterSubsystem;
+import frc.team2412.robot.subsystem.ShooterVisionSubsystem;
+import frc.team2412.robot.util.ShooterDataDistancePoint;
+
+public class IntakeBitmapCommand extends CommandBase {
+
+    public final double MISFIRE_VELOCITY = 400;;
+
+    public enum Bitmap {
```

Comment:
It looks like this doesn't cover all the possible cases - I'm guessing you omitted the ones where a ball was the correct color, but not detected by the proximity sensors (like `ingesthasball = false` and `ingestcorrectcolor = true`).

Unfortunately, the color sensors have a wider beam than the proximity sensors (or at least they should), which means it's possible that a cargo is detected by a color sensor but not by a proximity sensor, so we should handle cases like this.